src/text-overflow/TextOverflow.js

```diff
@@ -54,21 +64,35 @@ export default class TextOverflow extends React.Component {
     });
   };
 
+  getElementWidth = element => {
+    if (typeof window !== 'undefined') {
+      return parseFloat(window.getComputedStyle(element).width);
+    }
+    return null;
+  };
+
+  verifySize = () => {
+    const width = this.getElementWidth(ReactDOM.findDOMNode(this.refs.text));
+    if (!this.state.width) {
+      this.setState({ width });
+    } else if (width !== this.state.width) {
+      this.setState({ width }, this.reset); // trigger this.verifyOverflow
+    }
+  };
+
   verifyOverflow = () => {
     if (this.state.isOverflowing === false && typeof window !== 'undefined') {
       const text = ReactDOM.findDOMNode(this.refs.text);
       const textWithoutEllipsis = ReactDOM.findDOMNode(this.refs.textWithoutEllipsis);
 
-      if (text && textWithoutEllipsis) {
```

Comment:
this `if` shouldn't be necessary anymore ad I'm clearing `this.interval` and `this.timeout` in `componentWillUnmount`